src/internal/irmin-generics/irmin_generics.mli

```diff
@@ -0,0 +1,17 @@
+(*
+ * Copyright (c) 2016-2017 Thomas Gazagnaire <thomas@gazagnaire.org>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+module Type : module type of Type
```

Comment:
I'm not very fond of have this named `Irmin_generic.Type`. Why not stay consistent and call everything the same? e.g. `irmin-type`, `Irmin_type` and `irmin.type` (for the package name). I understand that `generics` is probably more correct than `type`, but `type` is already how it is named :-)

Reply:
The idea here was to create a namespace for future work to be used in the Irmin core but which is not necessarily exposed to the user via `Irmin.Type`, to give room to play with the API for optics. 

![image](https://user-images.githubusercontent.com/26125912/72349880-a6a93200-36dd-11ea-9584-f08057c2ed30.png)

I wasn't trying to come up with a more 'correct' name for `Irmin.Type`, which I think is already suitably named :slightly_smiling_face: We could go with `irmin-type` and `Irmin_type.Type`, but I consciously avoided that for sake of overloading `type` any further. `irmin-generics` is not a great name either; perhaps you have another suggestion?

Reply:
I was just thinking of having the type definitions at the toplevel of `Irmin_type` (e.g. inlinging `Type` in there).

Reply:
I'm not sure I understand you correctly, but inlining `Type` inside `Irmin_generics` means losing the ability to both keep the current `Type` interface and add new functionality.

Reply:
I thought about this a bit more and decided that I agree with you :slightly_smiling_face: The confusion around `generics` vs. `type` isn't worth any advantages it might bring. There's no immediate benefit of a `Irmin_generics` namespace and my ideas for its usefulness are vague at best. 

It'll be easy to introduce the indirection again later if it becomes necessary; it would be much harder to decide that it's time to inline if the indirection turns out not to be useful.

I made the changes you suggest in https://github.com/mirage/irmin/pull/921/commits/e185ddb117ac7bdb0f40c6ad8e71d88d02b89815. Let me know what you think.